.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,61 @@
+name: Unit tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  swiftlint:
+
+    name: Run SwiftLint
+
+    if: github.event_name == 'pull_request'
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out the code
+      uses: actions/checkout@v3
+    - name: Fetch PR's target branch
+      run: git fetch --no-tags --prune --depth=1 origin ${{ github.base_ref }}
+    - name: Run SwiftLint on files changed in the PR
+      uses: norio-nomura/action-swiftlint@3.2.1
+      with:
+        args: --force-exclude
+      env:
+        DIFF_BASE: ${{ github.base_ref }}
+
+  unit-tests:
+
+    name: Run unit tests
+
+    strategy:
+      matrix:
+        os: [macos-11, macos-12]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+    - name: Check out the code
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Run tests
+      # BSK does not support running tests in parallel, but the flag is required
```

Comment:
This could be slightly improved:
* The comment mentions BSK :)
* It looks like we can remove `--num-workers=1` argument as TRK tests seem to pass when run in parallel
* `${{ matrix.skipped_tests }}` can be removed since we don't define it anywhere